File: Level1/체육복.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = 0;
    answer = n - lost.size();
    sort(lost.begin(), lost.end());
    sort(reserve.begin(), reserve.end());
    vector<int>::iterator it;
    
    for(int i = reserve.size() - 1; i >= 0; i--)
    {
        if((it = find(lost.begin(),lost.end(), reserve.at(i))) != lost.end())
        {
            answer++;
            lost.erase(it);
            reserve.erase(find(reserve.begin(), reserve.end(), reserve.at(i)));
        }
    }
    
    while(!lost.empty() && !reserve.empty())
    {
        if((it = find(reserve.begin(),reserve.end(),lost.front() - 1)) != reserve.end() ||
                (it = find(reserve.begin(),reserve.end(),lost.front() + 1)) != reserve.end())
        {
            answer++;
            reserve.erase(it);
        }
        lost.erase(lost.begin());
    }
    return answer;
}

File: Level1/체육복.js
function solution(n, lost, reserve) {
    let answer = n - lost.length;

    lost = lost.sort((a, b) => {
        return a - b;
    });

    reserve = reserve.sort((a, b) => {
        return a - b;
    });

    for(let i = reserve.length - 1; i >= 0; i--){
        if(lost.indexOf(reserve[i]) !== -1){
            answer++;
            lost.splice(lost.indexOf(reserve[i]), 1);
            reserve.splice(reserve.indexOf(reserve[i]), 1);
        }
    }
    
    while(lost.length > 0 && reserve.length > 0){
        if(reserve.indexOf(lost[0] - 1) !== -1){
            answer++;
            reserve.splice(reserve.indexOf(lost[0] - 1), 1);
        }
        else if(reserve.indexOf(lost[0] + 1) !== -1){
            answer++;
            reserve.splice(reserve.indexOf(lost[0] + 1), 1);
        }
        lost.splice(0, 1);
    }
    
    return answer;
}